src/prefect/client/orchestration.py

```diff
@@ -3329,6 +3330,32 @@ async def read_resource_related_automations(
     async def delete_resource_owned_automations(self, resource_id: str):
         await self._client.delete(f"/automations/owned-by/{resource_id}")
 
+    async def api_version(self) -> str:
+        res = await self._client.get("/admin/version")
+        return res.json()
+
+    def client_version(self) -> str:
+        return prefect.__version__
+
+    async def api_compatible(self):
```

Comment:
`api_compatible` reads like a property name to me - would you mind updating this to `raise_for_api_version_mismatch` here and elsewhere? I think it makes it clearer what it's going to do when called (kind of like `raise_for_status`)

Reply:
done and done